src/Microsoft.ML.Data/DataLoadSave/Text/TextLoaderSaverCatalog.cs

```diff
@@ -5,58 +5,47 @@
 using Microsoft.ML.Runtime;
 using Microsoft.ML.Runtime.Data;
 using Microsoft.ML.Runtime.Data.IO;
-using Microsoft.ML.Runtime.Internal.Utilities;
-using Microsoft.ML.StaticPipe;
 using System;
-using System.Collections.Generic;
 using System.IO;
-using System.Linq;
-using System.Text;
 using static Microsoft.ML.Runtime.Data.TextLoader;
 
 namespace Microsoft.ML
 {
     public static class TextLoaderSaverCatalog
     {
-        /// <summary>
-        /// Create a text reader.
-        /// </summary>
-        /// <param name="catalog">The catalog.</param>
-        /// <param name="args">The arguments to text reader, describing the data schema.</param>
-        /// <param name="dataSample">The optional location of a data sample.</param>
-        public static TextLoader TextReader(this DataOperations catalog,
```

Comment:
Yesterday in the API design review, we decided against this approach. See the notes from the review here: [https://github.com/dotnet/apireviews/pull/81/files](https://github.com/dotnet/apireviews/pull/81/files)

But basically, the pattern decided will be:

1. Make a simple constructor/factory that has the most common parameters.
2. If there are advanced parameters that we don't want exposed in #1, then make another constructor/factory that takes the only the nested Arguments class (to be renamed to "Options").

We are going to move away from the `Action<Arguments> advancedSettings` approach. One main reason is because there can be conflicts between the "simple" parameters and the "advanced" parameters - and which one should win? Another reasoning is that it is simpler and understandable to construct and pass an object to a method.

I'd say, for this change, let's not move away from where we are going. You don't need to rename Arguments to Options.  But let's leave this overload, and remove the "advancedSettings" parameter below instead.

Reply:
I have left the overload as you suggested and removed the advanced arguments parameter from the constructors.

---
In reply to: [238009119](https://github.com/dotnet/machinelearning/pull/1784#discussion_r238009119) [](ancestors = 238009119)